Description: Revert upstream commit b0bce4ad41937dabf7e5c94dcce3caf4e88f3f97
 which introduced segfaults in many applications using
 g_file_make_directory_with_parents
Forwarded: not-needed
Bug-Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=680823
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/glib2.0/+bug/1035688

Index: b/gio/gfile.c
===================================================================
--- a/gio/gfile.c
+++ b/gio/gfile.c
@@ -3395,14 +3395,12 @@
       work_file = g_object_ref (parent_file);
 
       if (!result && my_error->code == G_IO_ERROR_NOT_FOUND)
-        {
-          g_clear_error (&my_error);
-          list = g_list_prepend (list, parent_file);  /* Transfer ownership of ref */
-        }
+        list = g_list_prepend (list, parent_file);  /* Transfer ownership of ref */
       else
         g_object_unref (parent_file);
     }
 
+  g_clear_error (&my_error);
   for (l = list; result && l; l = l->next)
     {
       result = g_file_make_directory ((GFile *) l->data, cancellable, &my_error);
